packages/terminal/src/browser/terminal-widget-impl.tsx

```diff
@@ -222,6 +234,12 @@ export class TerminalWidgetImpl extends TerminalWidget implements StatefulWidget
      * If id is provided attach to the terminal for this id.
      */
     async start(id?: number): Promise<number> {
+        const renderedItems: JSX.Element[] = [];
```

Comment:
we use the unidirectional data-flow pattern: all rendering should happen on each call of `onUpdateRequest` method based on the state, not here and there